crates/ink/codegen/src/generator/item_impls.rs

```diff
@@ -273,8 +273,7 @@ impl ItemImpls<'_> {
 
             #( #attrs )*
             #vis fn #checked_ident(#receiver #( , #wrapped_inputs )* ) -> #wrapped_output {
-                let #mut_token message_body = || { #( #statements )* };
-                ::core::result::Result::Ok(message_body())
+                ::core::result::Result::Ok(self.#ident( #( #input_bindings , )* ))
```

Comment:
Shouldn't it be the other way around: the unchecked calls the checked and just panics if an error is returned?

Reply:
It could be, but the thing is that the error case will never be hit. The only way to get to the error case is to go through the message dispatch mechanism - that's why we always wrap the body with `Ok`.

That said, if you have strong feelings about this I can change the unchecked implementation to always `expect()` the checked call.

Reply:
Sorry I was confusing this with the implementation in `ContractRef`. 

So now my question is: why generate this inherent impk in the first place? This is just generating the inherent impls so I don't think this will ever have enough information to generate a `LangError`.

We should lift this `Ok` as high as possible in the dispatch logic so it is obvious that at the moment that it always returns `Ok`. Because at the moment this has caused potential bug I mentioned in `dispatch` whereby the message returning `Err` gets wrapped in an `Ok` and the contract is not reverted.

Reply:
I didn't want to "lie" about the inherent return type and what was reflected in the ABI, which is why I wrapped the function return types and use that in the metadata generation.

After talking with Alex I realized that they had different consumers, so we could "lie" - but I didn't think of reverting those changes.

Either way, I've accepted your suggestion to move this further up the call stack, it makes sense